### PR TITLE
Updated Mac XCode Projects

### PR DESCRIPTION
**Pull request by KirbyKid256**
Brings the RSDKv3 mac builds up to speed with the latest GitHub commit. This includes updated icons, making sure the build doesn't crash for me, and being able to easily building signed frameworks for Ogg, Vorbis, and Theora.

The majority of the code is from @Sappharad's fork, although I did get permission beforehand to use it to do a pull request. I I have edited some of the changes down from the fork.

Please test this on Windows before you pull because while not much should have changed, I can't test it on Windows because I don't know how to compile it for Windows while on a Mac.

### CODE REVIEW

**Comment by KirbyKid256** [comment]
Realized this is the exact issue that caused it to crash for me #221

File: RSDKv3/Debug.hpp [hunk 1]
```diff
@@ -20,7 +20,14 @@ inline void PrintLog(const char *msg, ...)
         char pathBuffer[0x100];
 #if RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_UWP
         if (!usingCWD)
+#if RETRO_PLATFORM == RETRO_OSX
+        {
+            getResourcesPath(pathBuffer, sizeof(pathBuffer));
+            sprintf(pathBuffer, "%s/log.txt", pathBuffer);
```
Comment (Mefiresu) [inline RSDKv3/Debug.hpp:26]: Using the same buffer as both input and output for `sprintf` is unsafe and can cause issues, please use a temporary buffer for this.

File: dependencies/mac/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Mac
+
+Most of the Xcode project work has been done for you. Once you've set up the following sources in this dependencies folder, you can build RSDKv3 and the frameworks will compile beforehand.
+
+## Install dependencies
```
Comment (Mefiresu) [inline dependencies/mac/README.md:5]: Are .framework and manual zip downloads the best way to deal with dependencies on OSX? Something like homebrew or macports seems simpler but probably not the "Apple way" of doing things.

File: RSDKv3/Userdata.cpp [hunk 3]
```diff
@@ -287,8 +287,9 @@ void InitUserdata()
 #endif
 
 #if RETRO_PLATFORM == RETRO_OSX
-    sprintf(gamePath, "%s/RSDKv3", getResourcesPath());
-    sprintf(modsPath, "%s/RSDKv3/", getResourcesPath());
+    getResourcesPath(gamePath, sizeof(gamePath));
+    //sprintf(gamePath, "%s", getResourcesPath());
```
Comment (Mefiresu) [inline RSDKv3/Userdata.cpp:291]: You can remove this comment

**Review by Mefiresu — COMMENTED** [review]
Thanks for the PR!
The decomps now prefer CMake for building, as it makes the build process similar/simpler for all platforms (our Visual Studio solutions and Linux Makefiles are deprecated).
Would you be willing to make it work on MacOS?

File: RSDKv3/Userdata.cpp (hunk 3 above)
Comment (KirbyKid256) [reply RSDKv3/Userdata.cpp:291]: Forgot I left that in until I committed so sure

**Comment by KirbyKid256** [comment]
I ended up shooting myself in the foot by keeping a fork of a different fork of this repository, plus other errors and now I'll have to redo everything. I'll keep these in mind though.

File: RSDKv3/Debug.hpp (hunk 1 above)
Comment (KirbyKid256) [reply RSDKv3/Debug.hpp:26]: How would I make a temporary buffer?

**Pull request closed without merging**